WordPress/gutenberg
# Site Editor: Reduce the Amount of Data Passed Through the Components Tree

## Description

While working on https://github.com/WordPress/gutenberg/pull/26394 I've noticed that having a `TemplateNavigationItems` (plural) component, rendering lists of `NavigationItem`, prevented adding hooks to single items.
This led me to a bit of a simplification of the whole Site Editor navigation structure:

- Use a `TemplateNavigationItem` (singular) component instead, self-containing as much logic as possible, instead of inheriting from various hierarchical levels above.

In my initial exploration I've even tried to avoid passing `templates` through, relying on `getEntityRecord` in the `TemplateNavigationItem`.
This though had the side effect of firing a lot of unnecessary requests (one for the list, and one for each item). I wonder if there's a way to bypass the API and only select from Redux. 🤔 

## How has this been tested?

- Open the Site Editor and open the navigation sidebar.
- Navigate around the Templates and Template Parts menus.
- Make sure everything works as expected.
- Make sure the previews show up correctly.

## Types of changes

Code quality

## Checklist:
